src/reflect/scala/reflect/internal/tpe/TypeMaps.scala

```diff
@@ -215,15 +215,17 @@ private[internal] trait TypeMaps {
         this(sym.info)
     }
 
-    /** Called by mapOver to determine whether the original symbols can
-      *  be returned, or whether they must be cloned.
+    /** The index of the first symbol in `origSyms` which would have its info
+      * transformed by this type map.
       */
-    protected def noChangeToSymbols(origSyms: List[Symbol]): Boolean = {
-      @tailrec def loop(syms: List[Symbol]): Boolean = syms match {
-        case Nil     => true
-        case x :: xs => (x.info eq applyToSymbolInfo(x)) && loop(xs)
+    protected def firstChangedSymbol(origSyms: List[Symbol]): Int = {
+      @tailrec def loop(i: Int, syms: List[Symbol]): Int = syms match {
+        case Nil     => -1
+        case x :: xs =>
+          if (x.info eq applyToSymbolInfo(x)) loop(i+1, xs)
```

Comment:
`sym.info` is called here and immediately again in `applyToSymbolInfo`, which is an opportunity for a further saving.